src/commands.rs

```diff
@@ -862,9 +862,15 @@ pub fn command_json_arr_append<M: Manager>(
 
     // We require at least one JSON item to append
     args.peek().ok_or(RedisError::WrongArity)?;
-    let args = args
-        .map(|json| manager.from_str(&json.to_string_lossy(), Format::JSON))
-        .collect::<Result<_, _>>()?;
+
+    let args = args.try_fold::<_, _, Result<_, RedisError>>(
+        Vec::with_capacity(args.len()),
+        |mut acc, arg| {
+            let json = arg.try_as_str()?;
+            acc.push(manager.from_str(json, Format::JSON)?);
+            Ok(acc)
+        },
+    )?;
```

Comment:
- Can also do the same in `command_json_arr_insert`?
- I suggest to also test for failure with invalid utf-8 string such as `b'\x80abc'` (or other red cells in [table](https://en.wikipedia.org/wiki/UTF-8#:~:text=Red%20cells%20must%20never%20appear%20in%20a%20valid%20UTF-8%20sequence))